apps/web/src/components/borrow/table/DepositModal.tsx

```diff
@@ -0,0 +1,293 @@
+    const value = Number(e.target.value)
+    if (value > Number(data.balance) || value === 0) {
+      setAmountError(true)
+    } else {
+      setAmountError(false)
+    }
+    setAmount(value)
+  }
+  const tableColumns = {
+    action: {
+      header: 'Amount',
+      Cell: () => (
+        <ColumnLayout gap={0.5}>
+          <StackLayout>
+            <BaseInput
+              status={amountError ? 'error' : undefined}
+              onChange={handleInputCHange}
+              value={amount}
+            />
+          </StackLayout>
+          <Button
+            appearance="secondary"
+            variant="thin"
+            size="small"
+            className="u-fit-content-width"
+            onClick={() => setAmount(Number(data.balance))}
+          >
+            MAX
+          </Button>
+        </ColumnLayout>
+      ),
+      alignText: 'left',
+    },
+    asset: {
+      header: '',
+      Cell: ({ data: { symbol } }: CellProps) => <span>{symbol}</span>,
+      alignText: 'right',
+    },
+  }
+
+  const getValue = (index: number) => {
+    switch (index) {
+      case 0:
+        return (
+          <ColumnLayout align="flex-end" center>
+            <Text as="span" type="body 1" color="oxfordBlue">
+              <FormattedNumber value={data.supplyAPY} percent />
+            </Text>
+          </ColumnLayout>
+        )
+      case 1:
+        return (
+          <StackLayout>
+            <ColumnLayout align="flex-end" center>
+              <Icon name="union" size={16} color={themeContext.buttonDark} />
+              <Icon
+                name="arrow-right"
+                size={15}
+                color={themeContext.buttonDark}
+              />
+              <Text as="span" type="body 1" color="oxfordBlue">
+                1.00
+              </Text>
+            </ColumnLayout>
+            <Text as="span" type="body 1" color="oxfordBlue">
+              {'Liquidation at < 1.00'}
+            </Text>
+          </StackLayout>
+        )
+    }
+  }
+
+  const transactionColumns = {
+    action: {
+      header: 'Transaction overview',
+      Cell: ({ data: { name } }: CellProps) => (
+        <Text type="body 3" color="oxfordBlue" as="span">
+          {name}
+        </Text>
+      ),
+      alignText: 'left',
+    },
+    value: {
+      header: '',
+      Cell: ({ index }: CellProps) => getValue(index),
+      alignText: 'right',
+    },
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <StackLayout gap={5}>
+        <StackLayout gap={3}>
+          <Table
+            compact
+            columns={tableColumns}
+            data={[data]}
+            titleComponent={<Title />}
+          />
+          <Table compact columns={transactionColumns} data={transactionData} />
+        </StackLayout>
+
+        <BoxLayout gap={1.875}>
+          {hasError ? (
+            <CenterLayout>
+              <Icon name="error" size={75} />
```

Comment:
hm, I created a task for this part, when you see it, you can make it to DONE :), because it's fine to use it like you did just now.
Although we are still missing <Loader/> component.